### PR TITLE
[Draft/luci-interpreter]Add Quantize and Dequantize on TestUils

### PR DESCRIPTION
**Pull request by struss**
This draft commit add methods for Quantize and Dequantize on value to test Uint8 case.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/TestUtils.h [hunk 1]
```diff
@@ -58,6 +60,113 @@ template <typename T> std::vector<T> extractTensorData(const Tensor &tensor)
 std::vector<::testing::Matcher<float>> ArrayFloatNear(const std::vector<float> &values,
                                                       float max_abs_error = 1.0e-5f);
 
+template <typename T>
+inline std::vector<T> quantize(const std::vector<float> &data, float scale, int32_t zero_point)
+{
+  std::vector<T> q;
+  for (const auto &f : data)
+  {
+    if (getElementType<T>() == DataType::FLOAT32)
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/TestUtils.h:69]: How about removing this if statement and adding a below assertion at the beginning of this function? It seems that T should not be `float`.

E.g., `assert(getElementType<T>() != DataType::FLOAT32);`

File: compiler/luci-interpreter/src/kernels/Logistic.test.cpp [hunk 2]
```diff
@@ -25,33 +25,63 @@ namespace
...
   kernel.execute();
 
-  std::vector<float> ref_output_data{
-      0.5,      0.002473, 0.880797, 0.982014, //
-      0.952574, 0.119203, 0.999955, 0.731059, //
-  };
-  EXPECT_THAT(extractTensorData<float>(output_tensor),
-              ElementsAreArray(ArrayFloatNear(ref_output_data)));
-  // TODO make a Shape checking of output_tensor.
+  EXPECT_THAT(dequantize(extractTensorData<T>(output_tensor), output_tensor.scale(),
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Logistic.test.cpp:50]: Is it possible to separate this check code according to the `element_type`? E.g., not dequantizing values for float.

File: compiler/luci-interpreter/src/kernels/TestUtils.h [hunk 3]
```diff
@@ -58,6 +60,113 @@ template <typename T> std::vector<T> extractTensorData(const Tensor &tensor)
...
+  return q;
+}
+
+template <typename T>
+inline std::vector<float> dequantize(const std::vector<T> &data, float scale, int32_t zero_point)
+{
+  std::vector<float> f;
+  for (const T &q : data)
+  {
+    if (getElementType<T>() == DataType::FLOAT32)
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/TestUtils.h:85]: Similar to the above comment (replacing with assert).

File: compiler/luci-interpreter/src/kernels/Logistic.test.cpp [hunk 4]
```diff
@@ -25,33 +25,63 @@ namespace
...
+                   /*output_data=*/
+                   {
+                       0.5,      0.002473, 0.880797, 0.982014, //
+                       0.952574, 0.119203, 0.999955, 0.731059, //
+                       0.5,      0.002473, 0.880797, 0.982014, //
+                       0.952574, 0.119203, 0.999955, 0.731059, //
+                       0.5,      0.002473, 0.880797, 0.982014, //
+                       0.952574, 0.119203, 0.999955, 0.731059, //
+                   },
+                   /*input_min=*/-10.0f, /*input_max=*/10.0f, /*output_min=*/0.0f,
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Logistic.test.cpp:83]: How did you set these min/max values?

File: compiler/luci-interpreter/src/kernels/TestUtils.h (hunk 1 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/TestUtils.h]: Okay, I fix like what you suggested.

File: compiler/luci-interpreter/src/kernels/Logistic.test.cpp (hunk 2 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/Logistic.test.cpp]: Okay, I fix like what you suggested.

File: compiler/luci-interpreter/src/kernels/TestUtils.h (hunk 3 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/TestUtils.h]: Okay, I fix like what you suggested.

File: compiler/luci-interpreter/src/kernels/Logistic.test.cpp (hunk 4 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/Logistic.test.cpp:98]: I referenced from official Tensorflow example.

Comment (struss) [reply compiler/luci-interpreter/src/kernels/Logistic.test.cpp:99]: Oh, Found Something wired. I will make a new commit for this. Sorry for confused.

File: compiler/luci-interpreter/src/kernels/TestUtils.h [hunk 5]
```diff
@@ -58,6 +60,109 @@ template <typename T> std::vector<T> extractTensorData(const Tensor &tensor)
...
+{
+  if (getElementType<T>() == DataType::FLOAT32)
+  {
+    return {0.0, 0};
+  }
+  int32_t zero_point = 0;
+  float scale = 0;
+  const T qmin = std::numeric_limits<T>::min();
+  const T qmax = std::numeric_limits<T>::max();
+  const float qmin_double = qmin;
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/TestUtils.h:99]: Should be `double`, I guess.

Comment (struss) [reply compiler/luci-interpreter/src/kernels/TestUtils.h]: Okay! I will fix it.

**Comment by struss** [comment]
I will make a official PR for test util! thanks!

**Pull request closed without merging**